draft-ietf-quic-tls.md

```diff
@@ -684,19 +684,18 @@ MAY also include PADDING frames.
 ### Discarding Initial Keys
 
 Packets protected with Initial secrets ({{initial-secrets}}) are not
-authenticated, meaning that an attacker could spoof packets with the intent to
-disrupt a connection.  To limit these attacks, Initial packet protection keys
-can be discarded more aggressively than other keys.
-
-The successful use of Handshake packets indicates that no more Initial packets
-need to be exchanged, as these keys can only be produced after receiving all
-CRYPTO frames from Initial packets.  Thus, a client MUST discard Initial keys
-when it first sends a Handshake packet and a server MUST discard Initial keys
-when it first successfully processes a Handshake packet.  Endpoints MUST NOT
-send Initial packets after this point.
-
-This results in abandoning loss recovery state for the Initial encryption level
-and ignoring any outstanding Initial packets.
+authenticated, meaning that an attacker could spoof packets with the
+intent to disrupt a connection.  To limit these attacks, Initial
+packet protection keys can be discarded more aggressively than other
```

Comment:
You say this, but the only way it's more aggressive is ANY ACK at a higher encryption level causes discard of Initial keys.  I think it's worth outlining what the earliest point you can possibly drop the keys is?

If a client drops them upon processing a Handshake packet, I believe it needs to ensure the ACK frame sent in handshake is reliable in a sense, otherwise the server may never stop retransmitting the Initial data.

Possibly add a paragraph saying: "A peer MAY drop the Initial keys upon acknowledging a Handshake packet, but if it does so it must ensure the peer receives an ACK frame in a Handshake.  Otherwise, the peer will never stop transmitting Initial data."

Reply:
Yes, that sounds good.

Reply:
Sounds good to me too.

FWIW, I think that the other way would be to require endpoints to drop the Initial keys too when the handshake is confirmed, if the keys are not yet being dropped. But I think Ian's approach is better in terms of aligning the design.